emission/net/int_service/giles/timeout.py

```diff
@@ -0,0 +1,24 @@
+# Code comes from here: http://stackoverflow.com/questions/2281850/timeout-function-if-it-takes-too-long-to-finish
+
+from functools import wraps
+import errno
+import os
+import signal
+
+def timeout(seconds=10, error_message=os.strerror(errno.ETIME)):
```

Comment:
I am not sure we need to do this. Isn't there a built-in timeout in the requests library? Timing out network connections is a fairly common process...